gtsfm/utils/ellipsoid.py

```diff
@@ -0,0 +1,163 @@
+"""Algorithms to fit Ellisoid geometry to 3D points and extract the rotation matrix necessary to align point cloud with
+x, y, and z axes. Used in React Three Fiber Visualization Tool.
+
+Authors: Adi Singh
+"""
+
+import numpy as np
+from numpy.linalg import eig, inv
+from scipy.spatial import ConvexHull, convex_hull_plot_2d
+
+
+def center_point_cloud(pointsList: list) -> np.ndarray:
+    """Centers a point cloud with respect using mean values of x, y, and z.
+
+    Args:
+        pointCloud: list(length N) of lists(length 3) (representing the point cloud)
+
+    Returns:
+        Centered point cloud, of shape Nx3
+
+    Raises:
+        TypeError: if points list is not 3 dimensional
+    """
+    if len(pointsList[0]) != 3:
+        raise TypeError("Points list should be 3D")
+
+    points = np.array([np.array(p) for p in pointsList])
+
+    means = np.mean(points, axis=0)
+    points_centered = points - means
+    return points_centered
+
+
+def compute_convex_hull(points_np: np.ndarray) -> np.ndarray:
+    """Returns a set of vertices which represent the 3D Convex Hull (outer perimeter) of our point cloud.
+
+    Args:
+        points_np: point cloud of shape N x 3
+
+    Returns:
+        a reduced collection of points representing the convex hull, of shape M x 3 (M <= N)
+
+    Raises:
+        TypeError: if points_np is not 3 dimensional
+    """
+    if points_np.shape[1] != 3:
+        raise TypeError("Point Cloud should be 3D")
+
+    hull = ConvexHull(points_np)
+    hullSet = set()
+    for simplex in hull.simplices:
+        hullSet.add(tuple(points_np[simplex[0]]))
+        hullSet.add(tuple(points_np[simplex[1]]))
+
+    hull_set = np.array([np.array(p) for p in hullSet])
+    return hull_set
+
+
+def fit_ls_ellipsoid(hull_set: np.ndarray) -> np.ndarray:
```

Comment:
“LS” in the name is actually an implementation detail. It might also not be clear to everyone what it means. So I’d recommend calling this just “fit_ellipsoid”